lightning/src/util/macro_logger.rs

```diff
@@ -89,6 +90,29 @@ macro_rules! log_route {
 	}
 }
 
+pub(crate) struct DebugTx<'a>(pub &'a Transaction);
+impl<'a> std::fmt::Display for DebugTx<'a> {
+	fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
+		if self.0.input.len() == 1 && self.0.input[0].witness.last().unwrap().len() == 71 && (self.0.input[0].sequence >> 8*3) as u8 == 0x80 { write!(f, "commitment tx")?; }
+		else if self.0.input.len() == 1 && self.0.input[0].witness.last().unwrap().len() == 71 { write!(f, "closing tx")?; }
+		else if self.0.input.len() == 1 && self.0.input[0].witness.last().unwrap().len() == 133 && self.0.input[0].witness.len() == 5 { write!(f, "HTLC-timeout tx")?; }
```

Comment:
Don't these vary by +/- a few bytes?

Reply:
In fact that the line under, corrected to be right on both mainnet/testnet.

That's a basic logger, in case of aggregated inputs, it will only log first one